File: leetcode/0094-binary-tree-inorder-traversal/solution.cs
public class Solution {
    public IList<int> InorderTraversal(TreeNode root) {
        Stack<TreeNode> stack = new Stack<TreeNode>();
        List<int> result = new List<int>();

        while (root != null || stack.Count() > 0) {
            while (root != null) {
                stack.Push(root);
                root = root.left;
            }
            root = stack.Pop();
            result.Add(root.val);
            root = root.right;
        }
        return result;        
    }
}

File: leetcode/0094-binary-tree-inorder-traversal/solution.cpp
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        stack<TreeNode*> stack;
        vector<int> result;

        while (root != nullptr || !stack.empty()) {
            while (root != nullptr) {
                stack.push(root);
                root = root -> left;
            }

            root = stack.top();
            stack.pop();
            result.push_back(root -> val);
            root = root -> right;
        }
        return result;        
    }
};